src/CloudNative.CloudEvents.Avro/AvroEventFormatter.cs

```diff
@@ -41,30 +41,35 @@ public class AvroEventFormatter : CloudEventFormatter
         private const string DataName = "data";
 
         private static readonly string CloudEventAvroMediaType = MimeUtilities.MediaType + MediaTypeSuffix;
-        private static readonly RecordSchema avroSchema;
-        private static readonly DefaultReader avroReader;
-        private static readonly DefaultWriter avroWriter;
+        private readonly RecordSchema avroSchema;
+        private readonly IGenericRecordSerializer serializer;
         
-        static AvroEventFormatter()
+        /// <summary>
+        /// Creates an AvroEventFormatter using the default serializer.
+        /// </summary>
+        public AvroEventFormatter()
```

Comment:
I'd make this chain to the other constructor:

```csharp
public AvroEventFormatter() : this(new BasicGenericRecordSerializer(avroSchema))
```

(Of course, that only works if the change for schema parsing is reverted.)

Reply:
Suggestion applied